specification/cpim/resource-manager/Microsoft.AzureActiveDirectory/stable/2021-04-01/externalIdentities.json

```diff
@@ -0,0 +1,1167 @@
+              "readOnly": true,
+              "description": "The data from which the billing type took effect"
+            }
+          }
+        },
+        "tenantId": {
+          "type": "string",
+          "description": "An identifier of the Azure AD B2C tenant."
+        }
+      }
+    },
+    "B2CResourceSKU": {
+      "description": "SKU properties of the Azure AD B2C tenant. Learn more about Azure AD B2C billing at [aka.ms/b2cBilling](https://aka.ms/b2cBilling).",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the SKU for the tenant.",
+          "enum": [
+            "Standard",
+            "PremiumP1",
+            "PremiumP2"
+          ],
+          "x-ms-enum": {
+            "name": "B2CResourceSKUName",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "Standard",
+                "description": "Azure AD B2C usage is billed to a linked Azure subscription and uses a monthly active users (MAU) billing model."
+              },
+              {
+                "value": "PremiumP1",
+                "description": "Azure AD B2C usage is billed to a linked Azure subscription and uses number of authentications based billing."
+              },
+              {
+                "value": "PremiumP2",
+                "description": "Azure AD B2C usage is billed to a linked Azure subscription and uses number of authentications based billing."
+              }
+            ]
+          }
+        },
+        "tier": {
+          "type": "string",
+          "description": "The tier of the tenant.",
+          "enum": [
+            "A0"
+          ],
+          "x-ms-enum": {
+            "name": "B2CResourceSKUTier",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "A0",
+                "description": "The SKU tier used for all Azure AD B2C tenants."
+              }
+            ]
+          }
+        }
+      }
+    },
+    "NameAvailabilityResponse": {
+      "description": "Response of the CheckNameAvailability operation.",
+      "type": "object",
+      "properties": {
+        "message": {
+          "type": "string",
+          "description": "Description of the reason if name is not available."
+        },
+        "nameAvailable": {
+          "type": "boolean",
+          "description": "True if the name is available and can be used to create a new tenant. Otherwise false."
+        },
+        "reason": {
+          "$ref": "#/definitions/NameAvailabilityReason"
+        }
+      }
+    },
+    "NameAvailabilityReason": {
+      "description": "Describes the reason for the 'nameAvailable' value.",
+      "type": "string",
+      "enum": [
+        "AlreadyExists",
+        "Invalid"
+      ],
+      "x-ms-enum": {
+        "name": "NameAvailabilityReasonType",
+        "modelAsString": false,
```

Comment:
>false [](http://example.com/codeflow?start=25&length=5)

Consider set it to true to allow adding new enum values in the future without a breaking change.

[https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#use-extensible-enums](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#use-extensible-enums)